modules/c++/math.linear/source/Line2D.cpp

```diff
@@ -0,0 +1,228 @@
+/* =========================================================================
+ * This file is part of math.linear-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * math.linear-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <math/linear/Line2D.h>
+
+namespace math
+{
+namespace linear
+{
+Line2D::Line2D(const Point& P1, const Point& P2)
+{
+    const double dx = P2.row - P1.row;
+    const double dy = P2.col - P1.col;
+
+    if ((dy == 0) && (dx == 0))
+    {
+        throw except::Exception(Ctxt("Cannot create a line when P1 == P2"));
+    }
+    //Vertical if x values are the same
+    if (dx == 0)
+    {
+        mType = Line2D::VERTICAL;
+        mSlope = 0; // undefined
+        mXIntercept = P1.row;
+        mYIntercept = 0; // undefined
+    }
+    // Horizontal if y values are the same
+    else if (dy == 0)
+    {
+        mType = Line2D::HORIZONTAL;
+        mSlope = 0;
+        mYIntercept = P1.col;
+        mXIntercept = 0; // undefined
+    }
+    else {
+        mType = Line2D::NORMAL;
+        mSlope = dy / dx;
+        mYIntercept = P1.col - P1.row * mSlope; // Maybe a better way to do this. go back
+        mXIntercept = x(0.0);
```

Comment:
If this comment doesn't apply anymore, remove it.  Otherwise, was there another thing you were thinking about doing here?